crates/primitives/src/receipt.rs

```diff
@@ -223,10 +216,23 @@ impl ReceiptWithBloom {
     /// of the receipt:
     /// - EIP-1559, 2930 and 4844 transactions: `tx-type || rlp([status, cumulativeGasUsed,
     ///   logsBloom, logs])`
-    pub fn encode_enveloped(&self, out: &mut dyn bytes::BufMut) {
+    fn type_flag(&self) -> Option<u8> {
+        match self.receipt.tx_type {
+            TxType::Legacy => None,
+            tx_type => Some(tx_type as u8),
+        }
+    }
+
+    fn encode_2718_len(&self) -> usize {
+        self.as_encoder().length()
```

Comment:
not entirely sure this is correct @Rjected?

https://github.com/paradigmxyz/reth/blob/d4be773f5f71b49e7e40de3dd747a80af18daf74/crates/primitives/src/receipt.rs#L484-L498

Reply:
yes, using `as_encoder().length()` is not correct because the 2718 encoding does not have a header before the type byte. We should also create a test that checks that an encoded receipt actually has the same length as returned by `length`